codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2ConnectionEncoder.java

```diff
@@ -181,8 +183,8 @@ public ChannelFuture writeData(final ChannelHandlerContext ctx, final int stream
         }
 
         // Hand control of the frame to the flow controller.
-        ChannelFuture future =
-                flowController().sendFlowControlledFrame(ctx, stream, data, padding, endOfStream, promise);
+        ChannelFuture future = flowController().sendFlowControlledData(ctx, stream,
+                new FlowControlledData(ctx, streamId, data, padding, endOfStream, promise), promise);
```

Comment:
I think it might be cleaner to make interface:

    flowController().sendFlowControlledData(ctx, stream, data, padding, endOfStream, promise);

... and make `FlowControlledData` (and the other similar classes) inner classes of `DefaultHttp2RemoteFlowController`.  All the writing logic seems like it would be a better fit for the flow controller, since that's its responsibility.

Or am I missing a reason that these classes have to be within the encoder?